lib/rgb_matrix/engine.ex

```diff
@@ -50,62 +44,88 @@ defmodule RGBMatrix.Engine do
   end
 
   @doc """
-  Register a `RGBMatrix.Paintable` for the engine to paint pixels to.
-  This function is idempotent.
+  Register a paint function for the engine to send frames to. This function is
+  idempotent.
   """
-  @spec register_paintable(paintable :: module) :: :ok
-  def register_paintable(paintable) do
-    GenServer.call(__MODULE__, {:register_paintable, paintable})
+  @spec register_paintable(key :: any, paint_fn :: function) :: :ok
+  def register_paintable(key, paint_fn) do
```

Comment:
oh sorry, this isn't "key" as in a key on a keyboard, this is "please register this paintable function and here's a unique key to use to identify it".  This is so the caller can later unregister it by just passing the key back in.

Reply:
you were before implicitly using the paintable module as the map key; I made it so the caller has to supply it instead.

Reply:
Oh! Got it!

Instead of `key` then, how about `handle`? Although I may have a better idea here: https://github.com/ElixirSeattle/xebow/pull/56#discussion_r451061438